docs/contributor-docs/pnpm-migration-plan.md

```diff
@@ -0,0 +1,1579 @@
+
+### Monorepo Setup
+
+- **Lerna 8.1.9** for:
+  - Task running (`lerna run build --stream`)
+  - Package versioning (`lerna version --conventional-commits`)
+  - Package discovery (`lerna list --json`)
+- **npm workspaces** for dependency resolution
+- **Exact version pinning** for internal packages (e.g., `"@instructure/ui-view": "10.26.0"`)
+
+### Current Lerna Usage
+
+#### 1. Task Running (Easy to replace)
+
+| Before (Lerna)                               | After (pnpm)                    |
+| -------------------------------------------- | ------------------------------- |
+| `lerna run lint --stream`                    | `pnpm -r --stream lint`         |
+| `lerna run build --stream`                   | `pnpm -r --stream build`        |
+| `lerna run build:watch --stream`             | `pnpm -r --stream build:watch`  |
+| `lerna run bundle --stream --scope docs-app` | `pnpm --filter docs-app bundle` |
+
+#### 2. Package Discovery (Used by pkg-utils)
+
+- `packages/pkg-utils/lib/get-packages.js` calls `lerna list --json`
+- Returns package metadata (name, version, location)
+
+#### 3. Versioning & Changelog (CRITICAL)
+
+- `packages/ui-scripts/lib/utils/npm.js` uses `lerna version` with:
+  - `--conventional-commits` - Auto-generate changelog from git history
+  - `--exact` - No ^ in version ranges
+  - `--force-publish=*` - Bump ALL packages
+  - `--no-push` / `--no-git-tag-version` - Manual control
+
+### The Manual Version Problem
+
+**Current workflow when releasing:**
+
+```json
+---
+type: code
+---
+// Before release: ui-buttons/package.json
+{
+  "dependencies": {
+    "@instructure/ui-view": "10.26.0",
+    "@instructure/ui-utils": "10.26.0"
+  }
+}
+
+// After bumping to 10.27.0, Lerna updates:
+{
+  "dependencies": {
+    "@instructure/ui-view": "10.27.0",  // ← Manual update needed
+    "@instructure/ui-utils": "10.27.0"  // ← Manual update needed
+  }
+}
+```
+
+This happens across **101 packages** × ~10 internal deps each = **~1000 manual updates per release**.
+
+## Phase 1: pnpm + Workspace Protocol Migration
+
+**Goal:** Eliminate manual version updates, improve install performance, stricter dependencies
+
+**Risk:** Low
+
+### Step 1.1: Configuration Setup
+
+#### Create `pnpm-workspace.yaml`
+
+```yaml
+---
+type: code
+---
+packages:
+  - 'packages/*'
+```
+
+#### Update root `package.json`
+
+```json
+---
+type: code
+---
+{
+  "engines": {
+    "node": ">=22",
+    "npm": "Use pnpm instead.",
+    "pnpm": ">=9"
+  },
+  "pnpm": {
+    "overrides": {
+      "react": "^18",
+      "react-dom": "^18",
+      "@types/react": "^18"
+    }
+  }
+}
+```
+
+**Remove:**
+
+```json
+---
+type: code
+---
+{
+  "resolutions": {
+    // ← DELETE (replaced by pnpm.overrides)
+    "react": "^18",
+    "react-dom": "^18",
+    "@types/react": "^18"
+  }
+}
+```
+
+#### Create `.npmrc` for pnpm config
+
+```ini
+---
+type: code
+---
+# Strict peer dependencies (recommended)
```

Comment:
I think we also need [hoist-workspace-packages](https://pnpm.io/9.x/npmrc#hoist-workspace-packages) for `ui-scripts` to function